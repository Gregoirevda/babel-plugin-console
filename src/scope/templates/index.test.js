import templateHelpers from './';

describe('Templates', () => {
  const noop = () => {};

  describe('.buildGroupCollapsed', () => {
    it('calls template with correct string', () => {
      const template = jest.fn(() => noop);
      const { buildGroupCollapsed } = templateHelpers(template);
      buildGroupCollapsed('bob');
      expect(template).toHaveBeenCalledWith('console.groupCollapsed(title);');
    });

    it('invokes template with given title as an object', () => {
      const template = () => templateCurry;
      const templateCurry = jest.fn();
      const { buildGroupCollapsed } = templateHelpers(template);
      buildGroupCollapsed('matt');
      expect(templateCurry).toHaveBeenCalledWith({ title: 'matt' });
    });

    it('returns value from invoking template', () => {
      const template = () => () => 'hello world';
      const { buildGroupCollapsed } = templateHelpers(template);
      expect(buildGroupCollapsed('matt', 'woz', 'ere')).toBe('hello world');
    });
  });

  describe('.buildGroupEnd', () => {
    it('calls template with correct string', () => {
      const template = jest.fn(() => noop);
      const { buildGroupEnd } = templateHelpers(template);
      buildGroupEnd('bob');
      expect(template).toHaveBeenCalledWith('console.groupEnd(title);');
    });

    it('invokes template with given title as an object', () => {
      const template = () => templateCurry;
      const templateCurry = jest.fn();
      const { buildGroupEnd } = templateHelpers(template);
      buildGroupEnd('matt');
      expect(templateCurry).toHaveBeenCalledWith({ title: 'matt' });
    });

    it('returns value from invoking template', () => {
      const template = () => () => 'hello world';
      const { buildGroupEnd } = templateHelpers(template);
      expect(buildGroupEnd('matt', 'woz', 'ere')).toBe('hello world');
    });
  });

  describe('.buildLog', () => {
    it('calls template with correct string', () => {
      const template = jest.fn(() => noop);
      const { buildLog } = templateHelpers(template);
      buildLog('bob');
      expect(template).toHaveBeenCalledWith('console.log(args);');
    });

    it('invokes template with given args as an object', () => {
      const template = () => templateCurry;
      const templateCurry = jest.fn();
      const { buildLog } = templateHelpers(template);
      buildLog('matt', 'woz', 'ere');
      expect(templateCurry).toHaveBeenCalledWith({ args: ['matt', 'woz', 'ere'] });
    });

    it('returns value from invoking template', () => {
      const template = () => () => 'hello world';
      const { buildLog } = templateHelpers(template);
      expect(buildLog('matt', 'woz', 'ere')).toBe('hello world');
    });
  });
});
