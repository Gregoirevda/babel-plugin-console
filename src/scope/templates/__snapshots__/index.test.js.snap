// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates .buildGroupCollapsed calls template with correct string and indentation level: 0 1`] = `
"
    if (typeof window !== 'undefined') {
      console.groupCollapsed(title);
    } else {
      console.log('▼ ', title);
    }
  "
`;

exports[`Templates .buildGroupCollapsed calls template with correct string and indentation level: 1 1`] = `
"
    if (typeof window !== 'undefined') {
      console.groupCollapsed(title);
    } else {
      console.log('| ▼ ', title);
    }
  "
`;

exports[`Templates .buildGroupCollapsed calls template with correct string and indentation level: 2 1`] = `
"
    if (typeof window !== 'undefined') {
      console.groupCollapsed(title);
    } else {
      console.log('| | ▼ ', title);
    }
  "
`;

exports[`Templates .buildGroupCollapsed calls template with correct string and indentation level: 3 1`] = `
"
    if (typeof window !== 'undefined') {
      console.groupCollapsed(title);
    } else {
      console.log('| | | ▼ ', title);
    }
  "
`;

exports[`Templates .buildGroupEnd calls template with correct string 1`] = `
"
  if (typeof window !== 'undefined') {
    console.groupEnd(title);
  }
"
`;

exports[`Templates .buildLog calls template with correct string and indentation level: 1 1`] = `
"
    if (typeof window !== 'undefined') {
      console.log(args);
    } else {
      console.log('| | ', args);

    }
  "
`;

exports[`Templates .buildLog calls template with correct string and indentation level: 2 1`] = `
"
    if (typeof window !== 'undefined') {
      console.log(args);
    } else {
      console.log('| | | ', args);

    }
  "
`;

exports[`Templates .buildLog calls template with correct string and indentation level: 3 1`] = `
"
    if (typeof window !== 'undefined') {
      console.log(args);
    } else {
      console.log('| | | | ', args);

    }
  "
`;
